File: src/test/java/org/mastodon/mamut/StartMastodon.java
package org.mastodon.mamut;

import net.imagej.ImageJ;
import org.mastodon.mamut.launcher.MastodonLauncher;
import sc.iview.SciView;

public class StartMastodon {
	public static void main(String[] args) {
		try {
			//gives sciview/scenery a chance to run at all...
			SciView.xinitThreads();

			//without this, the GUI harvesting of plugins will not happen
			ImageJ ij = new ImageJ();
			ij.ui().showUI();

			//cannot use this one as it opens its own context...
			//new MastodonLauncherCommand().run();
			//
			//...thus copy&paste the run() above.. this time with the shared context
			final MastodonLauncher launcher = new MastodonLauncher( ij.getContext() );
			launcher.setLocationByPlatform( true );
			launcher.setLocationRelativeTo( null );
			launcher.setVisible( true );
		} catch (Exception e) {
			System.out.println("Got this exception: "+e.getMessage());
		}
	}
}

File: src/test/kotlin/org/mastodon/mamut/StartMastodon.kt
package org.mastodon.mamut

import graphics.scenery.SceneryBase
import graphics.scenery.utils.lazyLogger
import net.imagej.ImageJ
import org.mastodon.mamut.launcher.MastodonLauncher

object StartMastodon {
    private val logger by lazyLogger()
    @JvmStatic
    fun main(args: Array<String>) {
        try {
            //gives sciview/scenery a chance to run at all...
            SceneryBase.xinitThreads()

            //without this, the GUI harvesting of plugins will not happen
            val ij = ImageJ()
            ij.ui().showUI()

            //cannot use this one as it opens its own context...
            //new MastodonLauncherCommand().run();
            //
            //...thus copy&paste the run() above.. this time with the shared context
            val launcher = MastodonLauncher(ij.context)
            launcher.isLocationByPlatform = true
            launcher.setLocationRelativeTo(null)
            launcher.isVisible = true
        } catch (e: Exception) {
            logger.error("Got this exception: ${e.message}")
        }
    }
}
